File: app/src/main/java/com/ramez/shopp/adapter/MainSliderAdapter.java
package com.ramez.shopp.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.viewpager.widget.PagerAdapter;

import com.ramez.shopp.Classes.Constants;
import com.ramez.shopp.Classes.GlobalData;
import com.ramez.shopp.Classes.UtilityApp;
import com.ramez.shopp.Models.Slider;
import com.ramez.shopp.R;

import java.util.ArrayList;

public class MainSliderAdapter extends PagerAdapter {
    public ArrayList<Slider> sliderList;
    private Context context;
    private OnSliderClick onSliderClick;


    public MainSliderAdapter(Context context, ArrayList<Slider> sliderModels, OnSliderClick onSliderClick) {
        this.context = context;
        this.sliderList = sliderModels;
        this.onSliderClick=onSliderClick;
    }


    @Override
    public int getCount() {
        return sliderList.size();
    }


    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {
        LayoutInflater layoutInflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);

        View view = layoutInflater.inflate(R.layout.fragment_slider, container, false);

        String imageUrl;

        try {

            ImageView productImg = view.findViewById(R.id.slideImg);
            Slider slider=sliderList.get(position);

            if (UtilityApp.getLanguage().equals(Constants.English)) {
                imageUrl=slider.getImage();

            }
            else {
                imageUrl=slider.getImage2();

            }

            try {

                GlobalData.INSTANCE.GlideImgWeb(context,imageUrl,R.drawable.holder_image,productImg);

            } catch (Exception e) {
                e.printStackTrace();
            }



            container.addView(view);

            productImg.setOnClickListener(view1 -> {
                onSliderClick.onSliderClicked(position, slider);

            });

        } catch (Exception e) {
            e.printStackTrace();
        }

        return view;
    }


    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
        return view == object;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        View view = (View) object;
        container.removeView(view);
    }

    public interface OnSliderClick {
        void onSliderClicked(int position, Slider slider);
    }

}

File: app/src/main/java/com/amour/shop/adapter/MainSliderAdapter.kt
package com.amour.shop.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.viewpager.widget.PagerAdapter
import com.amour.shop.Models.Slider
import com.amour.shop.R
import com.amour.shop.classes.Constants
import com.amour.shop.classes.GlobalData.GlideImgWeb
import com.amour.shop.classes.UtilityApp
import java.util.ArrayList


class MainSliderAdapter(
    private val context: Context,
    var sliderList: MutableList<Slider?>?,
    private val onSliderClick: OnSliderClick
) :
        PagerAdapter() {
    override fun getCount(): Int {
        return sliderList?.size ?: 0
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val layoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val view = layoutInflater.inflate(R.layout.fragment_slider, container, false)
        val imageUrl: String
        try {
            val productImg = view.findViewById<ImageView>(R.id.slideImg)
            val slider = sliderList?.get(position)
            imageUrl = if (UtilityApp.getLanguage() == Constants.English) {
                slider?.image ?: ""
            } else {
                slider?.image2 ?: ""
            }
            try {
                GlideImgWeb(context, imageUrl, R.drawable.holder_image, productImg)
            } catch (e: Exception) {
                e.printStackTrace()
            }
            container.addView(view)
            productImg.setOnClickListener { view1: View? ->
                onSliderClick.onSliderClicked(
                    position,
                    slider
                )
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return view
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view === `object`
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        val view = `object` as View
        container.removeView(view)
    }

    interface OnSliderClick {
        fun onSliderClicked(position: Int, slider: Slider?)
    }
}